tracks/bash/exercises/acronym/mentoring.md

```diff
@@ -0,0 +1,103 @@
+### Reasonable Solutions
+
+```bash
+#!/usr/bin/env bash
+main () {    
+  # check to ensure an argument is supplied
+  INPUT="${@}"
+  if [[ -z "${INPUT}" ]]
+  then
+    echo "Please supply a string."
+    exit 1
+  else
+
+    # translate the punctuations into whitespace 
+    # Remove the apostrophe and anything but the alphabets
+    OUTPUT=$(echo "${INPUT}" | tr -d '\'\' | tr -c '[:alpha:]' ' ')
+
+    # Uppercase compatable with bash < 4.0
+    printf "%c" ${OUTPUT} | tr '[:lower:]' '[:upper:]'
+  fi
+}
+main "$@"
+
+#!/usr/bin/env bash
+main () {
+  [[ -n $1 ]] || exit 1
+  declare -ra WORDS=( ${1//[![:alpha:]]/ } )
+
+  for word in "${WORDS[@]}"; do
+    output+=${word:0:1}
+  done
+
+  # requires bash >= 4.0 to capitalize output
+  printf "%s\n" ${output^^}
+}
+main "$@"
+
+#!/usr/bin/env bash
+main () {
+  IFS=' !"#$%&()*+,-./:;<=>?@[\]^_`{|}~';
+  read -r -a array <<< "$*";
+  set acronym;
+  for element in "${array[@]}";
+  do
+      acronym+="${element:0:1}";
+  done
+  printf "%s\n" $acronym | tr '[:lower:]' '[:upper:]';
+  unset acronym;
+  unset IFS;
+}
+main "$@";
+```
+
+### Common Suggestions
+
+* Loop is not needed for the solution printf "%c" can be used to print the first characters of each element of an array. (see solution no.1) 
```

Comment:
Markdown backtick delimiters cn be used to show, i.e., `printf "%c"`, for when actual code is used in the suggestions..